Lib/test/test_pkgutil.py

```diff
@@ -489,6 +489,61 @@ def test_mixed_namespace(self):
     # XXX: test .pkg files
 
 
+class ExtendPathBaseTests(unittest.TestCase):
+    def test_input_type_invalid(self):
+        # path should be list instead of string
+        path = 'path'
+        name = 'foo'
+        self.assertEqual('path', pkgutil.extend_path(path, name))
+
+    def test_parent_package_raise_key_error(self):
+        path = ['path']
+        # sys.modules['foo'] raise KeyError
+        name = 'foo.bar'
+        self.assertEqual(['path'], pkgutil.extend_path(path, name))
+
+    def test_parent_package_raise_attr_error(self):
+        path = ['path']
+        # module 'datetime' has no attribute '__path__'
+        name = 'datetime.date'
+        self.assertEqual(['path'], pkgutil.extend_path(path, name))
+
+    def test_extend_path_files(self):
+        # create foo/bar.py
+        self.dirname = tempfile.mkdtemp()
+        sys.path.insert(0, self.dirname)
+        package_name = "foo"
+        package_path = os.path.join(self.dirname, package_name)
+        os.mkdir(package_path)
+        f = open(os.path.join(package_path, 'bar.py'), "w")
+        f.close()
+        # Search valid or invalid module
+        import foo
```

Comment:
This will leave `sys.modules` dirty from this test. Use the helper functions in `test.support` to make sure this method cleans up after itself.